pkg/util/iptables/iptables.go

```diff
@@ -328,10 +328,11 @@ func (runner *runner) checkRuleWithoutCheck(table Table, chain Chain, args ...st
 
 	// Sadly, iptables has inconsistent quoting rules for comments.
 	// Just unquote any arg that is wrapped in quotes.
-	argsCopy := make([]string, len(args))
-	copy(argsCopy, args)
-	for i := range argsCopy {
-		unquote(&argsCopy[i])
+	var argsCopy []string
+	for i := range args {
+		tmpField := args[i]
+		unquote(&tmpField)
+		argsCopy = append(argsCopy, strings.Fields(tmpField)...)
```

Comment:
I appreciate you bearing with me on this - I'm very wary of changes that I
can't explain.  I totally believe that there is a problem.

There's a debug log at line 358 - can you turn that on and send me the
output when it fails?  You can either make it unconditional or set V(1) or
just set --v=5 on your kube-proxy.

On Tue, Aug 18, 2015 at 5:47 PM, Jihoon Chung <notifications@github.com>
wrote:

> In pkg/util/iptables/iptables.go
> <https://github.com/kubernetes/kubernetes/pull/12839#discussion_r37370453>
> :
>
> > @@ -328,10 +328,11 @@ func (runner *runner) checkRuleWithoutCheck(table Table, chain Chain, args ...st
> >
> >  	// Sadly, iptables has inconsistent quoting rules for comments.
> >  	// Just unquote any arg that is wrapped in quotes.
> > -	argsCopy := make([]string, len(args))
> > -	copy(argsCopy, args)
> > -	for i := range argsCopy {
> > -		unquote(&argsCopy[i])
> > +	var argsCopy []string
> > +	for i := range args {
> > +		tmpField := args[i]
> > +		unquote(&tmpField)
> > +		argsCopy = append(argsCopy, strings.Fields(tmpField)...)
>
> Actually just the rules that have quoted multi-word comments get duped..
>
> and this is why I've unquoted and unpacked every word..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12839/files#r37370453>.
>